git_machete/client.py

```diff
@@ -551,6 +533,23 @@ def _slide_out_branches_from_git_machete(self,
             if self.up_branch[bd] != bu:
                 raise MacheteException(f"`{bu}` is not upstream of `{bd}`, cannot slide out")
 
+        self._modify_tree(branches_to_slide_out=branches_to_slide_out,
+                          opt_yes=opt_yes,
+                          opt_merge=opt_merge,
+                          opt_no_interactive_rebase=opt_no_interactive_rebase,
+                          opt_no_edit_merge=opt_no_edit_merge,
+                          opt_down_fork_point=opt_down_fork_point,
+                          delete_git_branches=delete_git_branches)
+
+    def _modify_tree(self,
```

Comment:
Hmmm isn't `modfiy_tree` too generic? It doesn't clearly encompass the action(s) that are actually being performed here

Reply:
I know, but there is a lot happening there and it's difficult to name it, any suggestions?

Reply:
Before that's decided... see the new comment wrt. rebases below

Reply:
(coz the decision wrt. these rebases may change how this method will finally be named)